### PR TITLE
EOS-16435 : Configure HAProxy as a part of S3 setup 3 Node VM component

### PR DESCRIPTION
**Pull request by shahidSTX**
Modified the haproxy config script to add 'N' instances based on env_type read from confstore. Tested the changes by running multiple iterations and invoking s3confstore API's to get pvt_ip and env_type.

### CODE REVIEW

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py [hunk 1]
```diff
@@ -113,11 +122,25 @@ def run():
...
+'''
+
+    target.write(header_text)
+    target.write(str1)
+    for i in range(0, n):
+        target.write(
+        "    server s3-instance-%s %s:%s check maxconn 110        # s3 instance %s\n"
+        % (i+1, pvt_ip, s3inport+i, i+1))
+    target.write(str2)
+    for i in range(0, n):
```
Comment (AjinkyaDhumal) [inline scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:137]: Auth can remain only one instance. No need to have a loop here

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py [hunk 2]
```diff
@@ -57,35 +61,42 @@ def run():
...
     cfg_file = '/etc/haproxy/haproxy.cfg'
     target = open(cfg_file, "a+")
-    target.write('''#-------S3 Haproxy configuration start--------------------------------
-#---------------------------------------------------------------------
+
+    n = 1
+    env = s3conf_store.get_config("cluster>env_type")
+
+    if env == 'VM':
+        n = 4
```
Comment (AjinkyaDhumal) [inline scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:69]: change the var n to numS3Instances

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py [hunk 3]
```diff
@@ -57,35 +61,42 @@ def run():
...
-#---------------------------------------------------------------------
-frontend main
+#----------------------------------------------------------------------
+frontend s3-main
     # s3 server port
     bind 0.0.0.0:80
     #bind 0.0.0.0:443 ssl crt /etc/ssl/stx/stx.pem
 
     option forwardfor
     default_backend app-main
```
Comment (AjinkyaDhumal) [inline scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:86]: Change app-main to s3-main as is the backend

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py (hunk 1 above)
Comment (shahidSTX) [reply scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:137]: Done

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py (hunk 2 above)
Comment (shahidSTX) [reply scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:69]: Done

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py (hunk 3 above)
Comment (shahidSTX) [reply scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:86]: Done

**Pull request closed without merging**